pybossa/api/__init__.py

```diff
@@ -154,8 +154,10 @@ def _retrieve_new_task(project_id):
 @crossdomain(origin='*', headers=cors_headers)
 def get_km_square(project_short_name=None):
     if project_short_name:
+        project = project_repo.get_by_shortname(project_short_name)
```

Comment:
Let's try to make as little requests to the database as possible. Here you are getting the project because you need it for task_repo.get_tasks_redundancy. But what does that last function use project for?

If you look at get_tasks_redundancy, it only needs the id. Do you have the id already? If yes, then just use it directly. If not, why not create a task_repo.get_tasks_redundancy_by_project_name function instead?

It's very important to limit the number of requests we do to the database, especially when this will be executed every time a user loads a new task!!